Feature/navbar setup

beratbaran40 opened the pull request.

beratbaran40 committed:
feat: Add string resources for bottom navigation bar items

beratbaran40 committed:
feat: Add category and selected category icons

beratbaran40 committed:
feat: Add selected search icon

beratbaran40 committed:
feat: Introduce AppDestination for navigation routes

Adds a sealed class `AppDestination` to define navigation destinations within the app. This class encapsulates properties for each screen, such as title, route, and icons for selected and unselected states.

It also includes a companion object to provide a list of bottom bar navigation items.

beratbaran40 committed:
feat: Implement bottom navigation bar

This commit introduces a bottom navigation bar in `MainActivity`.

- Adds `NavigationBar`, `NavigationBarItem`, and related Composables to the `Scaffold`.
- Implements navigation logic to switch between Home, Categories, and Favorites screens.
- The bottom bar is conditionally displayed only on top-level destinations.
- Configures item selection state, colors, icons, and click handlers to ensure correct navigation behavior, including state restoration.

beratbaran40 committed:
refactor: Capitalize route names in AppDestination

The route strings for all sealed `AppDestination` objects have been changed from lowercase to capitalized case to follow a consistent naming convention.

beratbaran40 committed:
feat: Capitalize bottom navigation bar route names

The route names used in the bottom navigation bar logic within MainActivity have been capitalized. This changes `home`, `categories`, and `favorites` to `Home`, `Categories`, and `Favorites` respectively to align with the `Screen` object's route definitions.

beratbaran40 committed:
refactor: Simplify boolean comparisons in BottomNavigationBar

The explicit `== true` checks in the `BottomNavigationBar` logic are redundant. This commit removes them to simplify the code and improve readability.

beratbaran40 committed:
refactor: Remove blank line from strings.xml

beratbaran40 committed:
feat: Refactor BottomBar into a reusable Composable

Extracted the bottom navigation bar logic from `MainActivity` into a new, reusable `LoopaBottomBar` composable. This simplifies the `MainActivity`'s Scaffold and improves modularity.

beratbaran40 committed:
feat: Add LoopaBottomBar composable for app navigation

cnrture commented on app/src/main/java/com/beratbaran/loopa/navigation/LoopaBottomBar.kt:
```diff
@@ -0,0 +1,95 @@
+
+                "Favorites" -> currentDestination?.hierarchy?.any {
+                    it.route?.substringBefore('?') == favoritesRoute
+                } == true
```
currentDestination?.hierarchy?.any {
                    it.route?.substringBefore('?') == favoritesRoute
                } == true

yukaridaki yorumlarim gecerli tek tek kontrol ediyoruz yine

beratbaran40 committed:
feat: Refactor LoopaBottomBar to retrieve items internally

The `LoopaBottomBar` composable is updated to manage its own navigation items, removing the need to pass them in from `MainActivity`. This simplifies the `MainActivity` and makes the bottom bar more self-contained.

beratbaran40 committed:
refactor: Simplify bottom bar navigation logic

Refactored the `LoopaBottomBar` composable to simplify the logic for determining when to show the bottom bar and which item is currently selected. This was achieved by using a map of top-level routes, making the code more concise and easier to maintain.

cnrture committed:
Merge branch 'main' into feature/navbar-setup
